zkevm-circuits/src/witness/tx.rs

```diff
@@ -340,25 +345,23 @@ impl Encodable for SignedTransaction {
 impl From<MockTransaction> for Transaction {
     fn from(mock_tx: MockTransaction) -> Self {
         let is_create = mock_tx.to.is_none();
-        let callee_address = match mock_tx.to {
-            Some(to) => to.address(),
-            None => Address::zero(),
-        };
         let sig = Signature {
             r: mock_tx.r.expect("tx expected to be signed"),
             s: mock_tx.s.expect("tx expected to be signed"),
             v: mock_tx.v.expect("tx expected to be signed").as_u64(),
         };
         let (rlp_unsigned, rlp_signed) = {
-            let legacy_tx = TransactionRequest::new()
+            let mut legacy_tx = TransactionRequest::new()
                 .from(mock_tx.from.address())
                 .nonce(mock_tx.nonce)
                 .gas_price(mock_tx.gas_price)
                 .gas(mock_tx.gas)
-                .to(callee_address)
                 .value(mock_tx.value)
                 .data(mock_tx.input.clone())
                 .chain_id(mock_tx.chain_id.as_u64());
+            if !is_create {
+                legacy_tx = legacy_tx.to(mock_tx.to.clone().map(|to| to.address()).unwrap());
```

Comment:
is the `clone` required?

Reply:
removed in https://github.com/scroll-tech/zkevm-circuits/pull/353/commits/7fd8036b2d31f74846b95abc06c786dd00c6062d